### PR TITLE
ID-5855: Temperature chart card title does not yet mention UTC time

### PR DESCRIPTION
**Pull request by nuria-balaguer**

### CODE REVIEW

File: src/LineChart/loadChart.js [hunk 1]
```diff
@@ -43,9 +43,38 @@ const loadChart = (chartData, elm, columns, options, isDateRange, onError) => {
 
             data.addRows(updatedChartData)
 
+            const dateFormat = new google.visualization.DateFormat({
+                formatType: 'medium', timeZone: 0,
+            })
+
+            dateFormat.format(data, 0)
+
+            const xTicks = []
```
Comment (dimaPortyanka) [inline src/LineChart/loadChart.js:52]: why not map? to add elements to xTicks?

Comment (nuria-balaguer) [reply src/LineChart/loadChart.js]: fixed

File: src/DownloadOptions/DownloadOptions.js [hunk 2]
```diff
@@ -57,39 +62,75 @@ const DownloadOptions = (props) => {
...
-                e[4],
+                e[0],
+                (loggerType === 'SKYCELL_SAVY_BLE' || isContainer) && e[1],
+                (loggerType === 'CARTASENSE') ? e[1] : e[4],
+                ...otherLoggers.map((element) => {
+                    let result = e[3 * pushedRows + 1]
+
+                    let pushedRowsIncrement = 1
+
+                    if (element.loggerType === 'SKYCELL_SAVY_BLE') {
```
Comment (dimaPortyanka) [inline src/DownloadOptions/DownloadOptions.js:90]: SKYCELL_SAVY_BLE move to constant

File: src/DownloadOptions/DownloadOptions.js [hunk 3]
```diff
@@ -57,39 +62,75 @@ const DownloadOptions = (props) => {
...
+                        ]
+
+                        pushedRowsIncrement = 2
+                    }
+
+                    pushedRows += pushedRowsIncrement
+                    return [
+                        element.value,
+                        result,
+                    ]
```
Comment (dimaPortyanka) [inline src/DownloadOptions/DownloadOptions.js:103]: can you briefly explain what is happening here?

File: src/DownloadOptions/DownloadOptions.js [hunk 4]
```diff
@@ -57,39 +62,75 @@ const DownloadOptions = (props) => {
...
 
-            return element.join(',')
+            return csvArray.join(',')
         }).join('\n')}`
 
+        const baseColumns = 'SERIAL_NUMBER,TIMESTAMP,AMBIENT_TEMPERATURE,INTERNAL_TEMPERATURE'
+
+        const columns = otherLoggers.reduce((acc, curr) => {
+            if (curr.loggerType === 'SKYCELL_SAVY_BLE') {
+                return `${acc},SERIAL_NUMBER,AMBIENT_TEMPERATURE,INTERNAL_TEMPERATURE`
```
Comment (dimaPortyanka) [inline src/DownloadOptions/DownloadOptions.js:114]: so are we always adding these elements to the current element? 
why not other.map(() => ).join() ?

File: src/DownloadOptions/DownloadOptions.js (hunk 3 above)
Comment (nuria-balaguer) [reply src/DownloadOptions/DownloadOptions.js:103]: The cartasense loggers only have internal data, the savy ones have internal and ambient, so I need to create the columns with this data: 
image This as example

File: src/DownloadOptions/DownloadOptions.js (hunk 4 above)
Comment (nuria-balaguer) [reply src/DownloadOptions/DownloadOptions.js:114]: all that gets joined to the baseColumns, but the baseColumns are only in the begining

File: src/DownloadOptions/DownloadOptions.js (hunk 2 above)
Comment (nuria-balaguer) [reply src/DownloadOptions/DownloadOptions.js]: fixed

File: src/DownloadOptions/DownloadOptions.js [hunk 5]
```diff
@@ -57,39 +67,75 @@ const DownloadOptions = (props) => {
...
+                    if (element.loggerType === SKYCELL_SAVY_BLE) {
+                        result = [
+                            e[3 * pushedRows + 1],
+                            e[3 * (pushedRows + 1) + 1],
+                        ]
+
+                        pushedRowsIncrement = 2
+                    }
+
+                    pushedRows += pushedRowsIncrement
```
Comment (dimaPortyanka) [inline src/DownloadOptions/DownloadOptions.js:104]: please extract pushedRows calculation outside of this function.

this map doing two different things split them

File: src/DownloadOptions/DownloadOptions.js (hunk 4 above)
Comment (nuria-balaguer) [reply src/DownloadOptions/DownloadOptions.js]: fixed

File: src/DownloadOptions/DownloadOptions.js (hunk 5 above)
Comment (nuria-balaguer) [reply src/DownloadOptions/DownloadOptions.js]: changed, check it please :)

File: src/DownloadOptions/__tests__/DownloadOptions.spec.js [hunk 6]
```diff
@@ -64,30 +95,50 @@ describe('shared-components/DownloadOptions', () => {
...
+        }).prop('onClick')()
+
+        setTimeout(() => {
+            expect(downloadDocument).toHaveBeenCalledWith({
+                headers: {
+                    'content-type': 'text/csv',
+                },
+                data: 'SERIAL_NUMBER,TIMESTAMP,AMBIENT_TEMPERATURE,INTERNAL_TEMPERATURE,SERIAL_NUMBER,AMBIENT_TEMPERATURE,INTERNAL_TEMPERATURE\n031-10279,16.08.2018 06:50,,24.7,bea4359jer9g9,24.7,21.5',
+            }, 'sensor_data_031-10279')
+            done()
```
Comment (dimaPortyanka) [inline src/DownloadOptions/__tests__/DownloadOptions.spec.js:141]: why set timeout is needed?

Comment (nuria-balaguer) [reply src/DownloadOptions/__tests__/DownloadOptions.spec.js]: fixed

**Review by dimaPortyanka — APPROVED** [review]

**Review by dimaPortyanka — APPROVED** [review]

**Pull request merged**